pol-core/pol/uoscrobj.cpp

```diff
@@ -4106,52 +4106,29 @@ BObjectImp* UBoat::script_method_id( const int id, Core::UOExecutor& ex )
   {
   case MTH_MOVE_OFFLINE_MOBILES:
   {
-    Core::xcoord newx;
-    Core::ycoord newy;
-    Core::zcoord newz;
-
     if ( ex.numParams() == 3 )
     {
-      if ( ex.getParam( 0, newx ) && ex.getParam( 1, newy ) &&
-           ex.getParam( 2, newz, Core::ZCOORD_MIN, Core::ZCOORD_MAX ) )
+      Core::Pos3d pos;
+      if ( ex.getPos3dParam( 0, 1, 2, &pos, realm() ) )
```

Comment:
did you consider inverting the if's here? might take even more nestings out:

if (!ex.getPos3d) { return error; } 
happy_path();